src/stash/src/upgrade/v15_to_v16.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::upgrade::MigrationAction;
+use crate::{StashError, Transaction, TypedCollection};
+
+pub mod objects_v15 {
+    include!(concat!(env!("OUT_DIR"), "/objects_v15.rs"));
+}
+pub mod objects_v16 {
+    include!(concat!(env!("OUT_DIR"), "/objects_v16.rs"));
+}
+
+/// Migration that grants the mz_introspection user USAGE privileges for the mz_system cluster.
+///
+/// Author - jkosh44 (ported by parkmycar)
+pub async fn upgrade(tx: &'_ mut Transaction<'_>) -> Result<(), StashError> {
+    const CLUSTERS_COLLECTION: TypedCollection<objects_v15::ClusterKey, objects_v15::ClusterValue> =
+        TypedCollection::new("compute_instance");
+
+    const MZ_SYSTEM_CLUSTER_NAME: &str = "mz_system";
+    const MZ_SYSTEM_ROLE_ID: objects_v15::RoleId = objects_v15::RoleId {
+        value: Some(objects_v15::role_id::Value::System(1)),
+    };
+    const MZ_INTROSPECTION_ROLE_ID: objects_v15::RoleId = objects_v15::RoleId {
+        value: Some(objects_v15::role_id::Value::System(2)),
+    };
+    const ACL_MODE_USAGE: objects_v15::AclMode = objects_v15::AclMode { bitflags: 1 << 8 };
```

Comment:
The hard-coding is actually by design! Say in the future we change the name of the system cluster, or how the `USAGE` flag is represented for `AclMode`. If in the migration path we depended on our common types, then a change like this could break the migration. By hard-coding these values we're effectively freezing the state of the Stash in time, which protects our migrations from our future-selves

Reply:
Is it possible that when we generate the `objects_vX` structs, that we can generate them with all the same methods/fields that the struct had at the time of the migration? That way we'd be able to call something like `objects_v15::AclMode::USAGE`, and we'd know it give us the result of what `USAGE` would have returned in v15?

I'm just worried that it will be way to easy to accidentally type  something like `bitflags: 1 << 7` and cause havoc.